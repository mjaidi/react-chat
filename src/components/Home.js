import React, { Component } from "react";
import Sidebar from "components/Sidebar";
import PleaseLogin from "components/PleaseLogin";

class Home extends Component {
  render() {
    const { isAuthenticated } = this.props.auth;
    if (isAuthenticated()) {
      return (
        <div className="home">
          <Sidebar client={this.props.client} history={this.props.history} />
        </div>
      );
    }
    return <PleaseLogin />;
  }
}

export default Home;
