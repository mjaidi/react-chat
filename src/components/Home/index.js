import React, { Component } from "react";

import UserList from "components/UserList";

class Home extends Component {
  login() {
    this.props.auth.login();
  }
  render() {
    const { isAuthenticated } = this.props.auth;
    if (isAuthenticated()) {
      return <UserList />;
    }
    return (
      <div className="container">
        <h4>You are not logged in! Please login to continue</h4>
      </div>
    );
  }
}

export default Home;
