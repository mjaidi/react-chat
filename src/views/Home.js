import React, { Component } from "react";
import Sidebar from "components/Sidebar";
import PleaseLogin from "components/PleaseLogin";
import { matchPath } from "react-router-dom";

class Home extends Component {
  render() {
    const { isAuthenticated } = this.props.auth;
    const hideSidebarForMobileConversations = () => {
      const match = matchPath(this.props.history.location.pathname, {
        path: "/home/conversations/:id",
        exact: true,
        strict: false
      });
      if (match !== null && window.innerWidth < 768) {
        return true;
      }
      return false;
    };

    if (isAuthenticated()) {
      return (
        <div className="home">
          {!hideSidebarForMobileConversations() && (
            <Sidebar
              client={this.props.client}
              history={this.props.history}
              activeConversation={this.props.history.location.id}
            />
          )}
        </div>
      );
    }
    return <PleaseLogin />;
  }
}

export default Home;
