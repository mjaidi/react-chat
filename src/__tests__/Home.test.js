import React from "react";
import { shallow } from "enzyme";
import Home from "components/Home";
import Auth from "../auth/auth0.js";

describe("Home", () => {
  let component;
  beforeEach(() => {
    component = shallow(<Home auth={new Auth()} />);
  });
  it("should render correctly without crashing", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render PleaseLogin component if authentication fails", () => {
    expect(component.find("PleaseLogin").exists()).toBe(true);
    expect(component.find("Sidebar").exists()).toBe(false);
  });

  it("should render Sidebar component if authentication succeeds", () => {
    component.setProps({ auth: { isAuthenticated: () => true } });
    expect(component.find("PleaseLogin").exists()).toBe(false);
    expect(component.find("Sidebar").exists()).toBe(true);
  });
});
