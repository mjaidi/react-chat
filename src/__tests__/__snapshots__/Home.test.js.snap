// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly without crashing 1`] = `<PleaseLogin />`;
